mapswipe_workers/mapswipe_workers/project_types/tile_map_service_grid/project.py

```diff
@@ -58,19 +59,30 @@ def validate_geometries(self):
             raise CustomError(f"Empty file. ")
 
         # check if more than 1 geometry is provided
-        elif layer.GetFeatureCount() > 1:
+        elif layer.GetFeatureCount() > featureNumber:
             logger.warning(
                 f"{self.projectId}"
                 f" - validate geometry - "
-                f"Input file contains more than one geometry. "
-                f"Make sure to provide exact one input geometry."
+                f"Input file contains more than {featureNumber} geometries. "
+                f"Make sure to provide less than {featureNumber} geometries."
             )
-            raise CustomError(f"Input file contains more than one geometry. ")
+            raise CustomError(f"Input file contains more than 10 geometries. ")
 
+        project_area = 0
+        geometry_collection = ogr.Geometry(ogr.wkbMultiPolygon)
         # check if the input geometry is a valid polygon
         for feature in layer:
             feat_geom = feature.GetGeometryRef()
             geom_name = feat_geom.GetGeometryName()
+            # add geometry to geometry collection
+            if geom_name == "MULTIPOLYGON":
+                for multi in feat_geom:
+                    geometry_collection.AddGeometry(multi)
+                dissolved_geometry = geometry_collection.UnionCascaded()
+                wkt_geometry_collection = dissolved_geometry.ExportToWkt()
```

Comment:
the `UnionCascaded` function should just be called if the input geometry is a multipolygon, if I put the `UnionCascaded` function completely outside of the loop, it would be called for single polygon as well (although not necessary), might lead to additional processing time

Reply:
I'm not sure. There could also be a geojson file which contains 2 single polygons that are overlapping and then the Cascaded Union function should be called.